apps/omg/lib/omg/ethereum_event_listener/core.ex

```diff
@@ -70,89 +63,29 @@ defmodule OMG.EthereumEventListener.Core do
           {:dont_fetch_events, t()} | {:get_events, {non_neg_integer, non_neg_integer}, t()}
-  def get_events_range_for_download(%__MODULE__{cached: %{events_upper_bound: upper}} = state, %SyncGuide{
-        sync_height: sync_height
-      })
-      when sync_height <= upper,
-      do: {:dont_fetch_events, state}
-
-  @decorate span(service: :ethereum_event_listener, type: :backend, name: "get_events_range_for_download/2")
-  def get_events_range_for_download(
-        %__MODULE__{
-          cached: %{request_max_size: request_max_size, events_upper_bound: old_upper_bound} = cached_data
-        } = state,
-        %SyncGuide{root_chain_height: root_chain_height, sync_height: sync_height}
-      ) do
-    # grab as much as allowed, but not higher than current root_chain_height and at least as much as needed to sync
-    # NOTE: both root_chain_height and sync_height are assumed to have any required finality margins applied by caller
-    next_upper_bound =
-      min(root_chain_height, old_upper_bound + request_max_size)
-      |> max(sync_height)
-
-    new_state = %__MODULE__{
-      state
-      | cached: %{cached_data | events_upper_bound: next_upper_bound}
-    }
+  def get_events_range(state, sync_guide) do
+    case sync_guide.sync_height <= state.synced_height do
+      true ->
+        {:dont_fetch_events, state}
 
-    {:get_events, {old_upper_bound + 1, next_upper_bound}, new_state}
-  end
-
-  @doc """
-  Stores the freshly fetched ethereum events into a memory-cache
-  """
-  @decorate span(service: :ethereum_event_listener, type: :backend, name: "add_new_events/2")
-  @spec add_new_events(t(), list(event)) :: t()
-  def add_new_events(
-        %__MODULE__{cached: %{data: data} = cached_data} = state,
-        new_events
-      ) do
-    %__MODULE__{state | cached: %{cached_data | data: data ++ new_events}}
-  end
-
-  @doc """
-  Pop some ethereum events stored in the memory-cache, up to a certain height
-  """
-  @decorate span(service: :ethereum_event_listener, type: :backend, name: "get_events/2")
-  @spec get_events(t(), non_neg_integer) :: {:ok, list(event), list(), non_neg_integer, t()}
-  def get_events(
-        %__MODULE__{synced_height_update_key: update_key, cached: %{data: data}} = state,
-        new_sync_height
-      ) do
-    {events, new_data} = Enum.split_while(data, fn %{eth_height: height} -> height <= new_sync_height end)
+      _ ->
+        # if sync_guide.sync_height has applied margin (reorg protection)
+        # the only thing we need to be aware of is that we don't go pass that!
+        # but we want to move as fast as possible so we try to fetch as much as we can (request_max_size)
 
-    new_state =
-      state
-      |> Map.update!(:synced_height, &max(&1, new_sync_height))
-      |> Map.update!(:cached, &%{&1 | data: new_data})
-      |> struct!()
+        next_upper_bound = min(sync_guide.sync_height, state.synced_height + 1 + state.request_max_size)
```

Comment:
Can you explain the significance of `state.synced_height + 1`? Perhaps we should name this value since it's used later on [L88](https://github.com/omgnetwork/omg-childchain-v1/pull/27/files#diff-7bb6c265b84427cdc213b394cac4180864191e7b263d335eb2a285576d8e1e16R88)